### DISCUSSION TITLE
decoding the raw data from Live Timing Client

### DISCUSSION THREAD

**Discussion by w-i-z-a-r-d** [theOehrly/Fast-F1#604] [Q&A]
the raw data collected how is it encrypted, base64?

**Comment by theOehrly** [comment]
It's not really "encrypted" but yes, base64 and some compression. This is the code for decoding it:

https://github.com/theOehrly/Fast-F1/blob/70c038afabcc3cd5285a793e1014a36cfc9d1ffa/fastf1/_api.py#L1756-L1758

**Reply by w-i-z-a-r-d** [reply]
sometimes in the collected data there is a Key "G", what does that contain ?

**Reply by theOehrly** [reply]
Some kind of "groups token – an encrypted string representing group membership", so it is only related to the SignalR connection itself. This https://blog.3d-logic.com/2015/03/29/signalr-on-the-wire-an-informal-description-of-the-signalr-protocol/ seems to be the easiest to read description of the SignalR protocol that I have found so far.
